### ISSUE TITLE
[Feature request]: sbb-dialog should announce changes of accessibilityLabel in overlay live region

### ISSUE THREAD

**Issue by georgwittberger-telekom-mms** [sbb-design-systems/lyne-components#3982]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/orgs/sbb-design-systems/projects/4) for an issue that matches the one I want to file, without success.

### Request type

Request for enhancement of a component

### It affects the following packages

Components

### Functionality

**Situation:**

We are using the `sbb-dialog` component in such a way that requires us to change the title text of the dialog while it is already open. Think of it as switching views inside the dialog.

Currently, the live region component `sbb-screen-reader-only` created by `sbb-dialog` only announces the dialog title right after the dialog is opened. Subsequent changes of the title - no matter if text content of `sbb-dialog-title` is updated or property `accessibilityLabel` is changed - are not announced which makes it hard for screen reader users to recognize that the view has changed.

**Proposal:**

Everytime the property `accessibilityLabel` is updated with a value that differs from the current one this new label text should be set in the live region to announce this new dialog title.

The reset of the live region can stay as is, i.e. on any interaction within the dialog the live region is cleared.

**Workaround:**

We have implemented a workaround in our Next.js project which involves updating the live region using a ref callback and some Shadow DOM manipulation. If someone is interested in that code, drop a comment.
